lib/sidekiq/config.rb

```diff
@@ -41,9 +41,17 @@ class Config
       l.warn(ex.backtrace.join("\n")) unless ex.backtrace.nil?
     }
 
+    DEPRECATION_HANDLER = ->(msg, ctx, caller) {
+      logger = Sidekiq.default_configuration.logger
+
+      line = caller[0]
+      logger.warn "DEPRECATION WARNING: #{msg} (called from #{line})"
```

Comment:
(On Masto where @mperham's toot was mis-rendered for me as a poll and I responded instead of coming here)

A "print out a deprecation warning" message like this is challenging in two ways:
* It does not include the name of the gem producing this, so it can require grepping the bundle-installed gems to figure out where it's coming from
* The user of `self.deprecate` is not encouraged (or allowed?) to provide a solution to address the deprecation.

If I were making this more general, I would do:

```ruby
def self.deprecate(problem:, solution:)
  default.configuration.handle_deprecation(problem: problem, solution: solution, caller: caller)
end

# ...

DEPRECATION_HANDLER => ->(problem, solution, context, caller) {
  logger = Sidekiq.default_configuration.logger

  line = caller[0]
  logger.warn("[Sidekiq] DEPRECATION: #{problem}: This can be avoided by #{solution}")
}
```

Or maybe something better, but seeing this in my logs I would a) know it was about Sidekiq and b) be given some more info about what to do about removing the deprecated stuff I am using.  I realize the caller of `deprecate` can given a non-helpful `solution`, but it would at least encourage helpfulness.

Reply:
I think adding `logger.warn("[Sidekiq]...")` (the [Sidekiq] prefix), would be kind of an antipattern, right? 

Wouldn't that (imho) belong in the logger itself? 
`logger = MyLoggerClass.new(prefix: '[Sidekiq'])`, instead of the outputter?

The output should be the raw message imo